### DISCUSSION TITLE
feat: Allow Pinning Models in the Model Selection Dropdown

### DISCUSSION THREAD

**Discussion by bannert1337** [open-webui/open-webui#3443] [General]
**Is your feature request related to a problem? Please describe.**
As a user studying for a master's degree in computer science, I often need to switch between several specific models frequently. It becomes frustrating to scroll through an extensive list of models every time I need to switch between them.

**Describe the solution you'd like**
I would like the ability to pin certain models to the top of the model selection dropdown. This would ensure that my frequently used models are easily accessible without the need to scroll through the entire list.

**Describe alternatives you've considered**
An alternative solution could be to introduce a "favorites" section within the dropdown where users can mark certain models as favorites and have them appear in a prioritized section at the top.

**Additional context**
This feature would significantly enhance the user experience for those who routinely work with multiple models, especially in academic settings where efficiency is crucial. Additionally, the pinned state could be saved in user preferences to maintain consistency across sessions.

**Comment by silentoplayz** [comment]
Model files can be re-organized via drag-&-drop on laptop and PC devices in the `Workspace` > `Models` section of Open WebUI. You can technically use this to "pin" your frequently used and favored models to be available at the top of the model selection dropdown.

**Reply by gschintgen** [reply]
Thanks, that's helpful but not a complete solution. (I have the same - minor - UI issue as OP.)
In particular when new models are coming out it's cumbersome to move the older ones back out of the way instead of just unpinning them (removing from favorites).

Also, when customising a model, i.e. adding a "new" model with a custom system prompt, the dropdown list for the base model doesn't seem to be sorted at all. At least it's not in alphabetical order. This is quite an unwieldy list if you have all the models provided by OpenRouter and OpenAI in there.

**Reply by dan-passaro** [reply]
I'm unable to see any models at all when I go to "Workspace -> Models". It looks like I have to create a model myself for it to show up there. Being able to pin models to the top of the model dropdown by starring them or something like that would be a great feature imo; I connected Open WebUI to OpenRouter and now my model selection dropdown is a bit of a pain to use.

**Reply by BenAgin1** [reply]
@dan-passaro you can reorder the model order by going to settings>models. still pretty cumbersome to drag a model all the way to the top, but it works! I also had the same problem of no models in that workspace tab
